Build breaks when using GOFLAGS="-p=1"

> /kind bug


**What happened**:
When building kubernetes from source for 1.10 it was possible to limit the parallel linker calls using GOFLAGS="-p 1". (I understand that the new format is GOFLAGS="-p=1" so I am using that)
With kubernetes 1.11 the build aborts:

```
go test: p flag may be set only once
run "go help test" or "go help testflag" for more information
!!! [1021 19:40:30] Call tree:
!!! [1021 19:40:30]  1: /home/tkruse/kubernetes/src/kubernetes-1.12.0/hack/lib/golang.sh:735 kube::golang::build_binaries_for_platform(...)
!!! [1021 19:40:30]  2: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
!!! [1021 19:40:30] Call tree:
!!! [1021 19:40:30]  1: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
!!! [1021 19:40:30] Call tree:
!!! [1021 19:40:30]  1: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
make: *** [Makefile:92: all] Error 1
==> ERROR: A failure occurred in build().
    Aborting...
```

**What you expected to happen**:
The build should complete, using just one process for linking.

**How to reproduce it (as minimally and precisely as possible)**:
export GOFLAGS="-p=1"
make -j1

**Anything else we need to know?**:
The environment is quite memory constrained, therefore preventing the build to fork a linker for each CPU is required. Any other option instead of the GOFLAGS would work as well.

**Environment**:
- Kubernetes version (use `kubectl version`):\
1.12.0
- OS (e.g. from /etc/os-release):
Arch Linux ARM

- Others:
go version go1.11.1 linux/arm64

--- a/hack/lib/golang.sh
+++ b/hack/lib/golang.sh
@@ -677,9 +677,7 @@ kube::golang::build_binaries() {
     local host_platform
     host_platform=$(kube::golang::host_platform)
 
-    # Use eval to preserve embedded quoted strings.
     local goflags goldflags goasmflags gogcflags
-    eval "goflags=(${GOFLAGS:-})"
     goldflags="${GOLDFLAGS:-} -s -w $(kube::version::ldflags)"
     goasmflags="-trimpath=${KUBE_ROOT}"
     gogcflags="${GOGCFLAGS:-} -trimpath=${KUBE_ROOT}"

--- a/hack/make-rules/test.sh
+++ b/hack/make-rules/test.sh
@@ -171,7 +171,6 @@ done
 shift $((OPTIND - 1))
 
 # Use eval to preserve embedded quoted strings.
-eval "goflags=(${GOFLAGS:-})"
 eval "testargs=(${KUBE_TEST_ARGS:-})"
 
 # Used to filter verbose test output.

--- a/hack/make-rules/vet.sh
+++ b/hack/make-rules/vet.sh
@@ -35,9 +35,6 @@ fi
 # See https://github.com/golang/go/issues/16086 for details.
 go install ./cmd/...
 
-# Use eval to preserve embedded quoted strings.
-eval "goflags=(${GOFLAGS:-})"
-
 # Filter out arguments that start with "-" and move them to goflags.
 targets=()
 for arg; do

